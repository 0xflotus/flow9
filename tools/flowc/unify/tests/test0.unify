// Tests all the basic types
import maybe;

α0 = int;

α1 c= α0;

α2 c= bool;
bool c= α3;

α4 c= int;
α5 c= double;
α6 c= string;

α7 c= native;
native c= α8;

α9 c= flow;
flow c= α10;

α11 c= void;
void c= α12;

α13 c= ?;
? c= α14;
α15 c= ??;
?? c= α16;

α17 c= () -> void;
() -> void c= α18;

α19 c= [int];
[int] c= α20;

α21 c= ref int;
ref int c= α22;

α23 c= None<?>;
None<?> c= α24;

α25 c= union Maybe<?>;
union Maybe<?> c= α26;

α27 c= (bool, int, flow, ref string, [native], (?) -> ??) -> double;
(bool, int, flow, ref string, [native], (?) -> ??) -> double c= α28;

α29 c= (None<?>, (Some<??>) -> None<?>) -> Some<??>;
(None<?>, (Some<??>) -> None<?>) -> Some<??> c= α30;

α31 c= (union Maybe<?>) -> void;
(union Maybe<?>) -> void c= α32;

α33 c= () -> union Maybe<?>;
() -> union Maybe<?> c= α34;

α35 c= (None<?>) -> void;
(None<?>) -> void c= α36;

α37 c= () -> None<?>;
() -> None<?> c= α38;

α39 c= ref None<?>;
ref None<?> c= α40;

α41 c= ref union Maybe<?>;
ref union Maybe<?> c= α42;
