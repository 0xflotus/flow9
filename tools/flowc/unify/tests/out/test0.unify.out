α0=int
α1=int
α2=bool
α3=bool
α4=int
α5=double
α6=string
α7=native
α8=native
α9=flow
α10=flow
α11=void
α12=void
α13=?
α14=?
α15=??
α16=??
α17=() -> void
α18=() -> void
α19=[int]
α20=[int]
α21=ref int
α22=ref int
α23=None<?>
α24={None<?> .. *}
α25={* .. Maybe<?>}
α26={Maybe<?> .. *}
α27=(bool, int, flow, ref string, [native], (?) -> ??) -> double
α28=(bool, int, flow, ref string, [native], (?) -> ??) -> double
α29=({None<?> .. *}, (Some<??>) -> {None<?> .. *}) -> Some<??>
α30=(None<?>, ({Some<??> .. *}) -> None<?>) -> {Some<??> .. *}
α31=({Maybe<?> .. *}) -> void
α32=({* .. Maybe<?>}) -> void
α33=() -> {* .. Maybe<?>}
α34=() -> {Maybe<?> .. *}
α35=({None<?> .. *}) -> void
α36=(None<?>) -> void
α37=() -> None<?>
α38=() -> {None<?> .. *}
α39=ref None<?>
α40=ref {None<?> .. *}
α41=ref {* .. Maybe<?>}
α42=ref {Maybe<?> .. *}
α100=void
α101=void
α102=int
α103=int
α104=int
α105=int
α106=int
α107=int
α108=double
α109=bool
α110=int
α111=flow
α112=ref string
α113=[native]
α114=(?) -> ??
α115=string
α116=string
α117=native
α118=??
α119=?
α120=double
α121=bool
α122=int
α123=flow
α124=ref string
α125=[native]
α126=(?) -> ??
α127=string
α128=string
α129=native
α130=??
α131=?
α132=Some<??>
α133={None<?> .. *}
α134=(Some<??>) -> {None<?> .. *}
α135={None<?> .. *}
α136=Some<??>
α137={Some<??> .. *}
α138=None<?>
α139=({Some<??> .. *}) -> None<?>
α140=None<?>
α141={Some<??> .. *}
α142=void
α143={Maybe<?> .. *}
α144=void
α145={* .. Maybe<?>}
α146={* .. Maybe<?>}
α147={Maybe<?> .. *}
α148=void
α149={None<?> .. *}
α150=void
α151=None<?>
α152=None<?>
α153={None<?> .. *}
α154=None<?>
α155=None<?>
α156={None<?> .. *}
α157={None<?> .. *}
α158={* .. Maybe<?>}
α159={* .. Maybe<?>}
α160={Maybe<?> .. *}
α161={Maybe<?> .. *}
